pkgs/development/libraries/sparrow3d/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, fetchFromGitHub
+, pkg-config
+, SDL
+, SDL_image
+, SDL_mixer
+, SDL_net
+, SDL_ttf
+, stdenv
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sparrow3d";
+  version = "unstable-2020-10-06";
+
+  outputs = [ "out" "dev" ];
+
+  src = fetchFromGitHub {
+    owner = "theZiz";
+    repo = "sparrow3d";
+    rev = "2033349d7adeba34bda2c442e1fec22377471134";
+    hash = "sha256-28j5nbTYBrMN8BQ6XrTlO1D8Viw+RiT3MAl99BAbhR4=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  propagatedBuildInputs = [
```

Comment:
```suggestion
  buildInputs = [
```

Reply:
changing `propagatedBuildInputs` to `buildInputs` here causes `nix buld '.#hase'` to fail with:
```
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Package sdl was not found in the pkg-config search path.
Perhaps you should add the directory containing `sdl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sdl', required by 'sparrow3d', not found
```

sparrow both links to sdl and when a project like Hase `#include <sparrow3d.h>`s that in turn `#include <SDL.h>`s. i think `propagatedBuildInputs` is the solution here, though maybe the sparrow3d.pc pkg-config file could `-Isystem ${sdl.dev}/include/SDL` & friends as an alternative, if you think that's better?

Reply:
Well, two things:

1. I believe you can use the function `makePkgconfigItem` to generate pkg-config `.pc` files;
2. I don't know how other distros handle things like this. _However_, given that other distros just throw everything in a global namespace, they are not useful to draw a line here.

OK, keep `propagated???`

Reply:
switched to `makePkgconfigItem` (thanks for letting me know about it!) & kept `propagatedBuildInputs`